include/aikido/constraint/dart/IKRankingStrategy.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef AIKIDO_CONSTRAINT_DART_IKRANKINGSTRATEGY_HPP_
+#define AIKIDO_CONSTRAINT_DART_IKRANKINGSTRATEGY_HPP_
+
+#include <dart/dynamics/dynamics.hpp>
+#include "aikido/statespace/dart/MetaSkeletonStateSpace.hpp"
+
+namespace aikido {
+namespace constraint {
+namespace dart {
+
+class IKRankingStrategy
+{
+public:
+  // TODO (avk): Move this to protected?
+  // Read this in detail:
+  // 1. https://stackoverflow.com/questions/2290733/
+  // 1. initialize-parents-protected-members-with-initialization-list-c
+  // 2. https://stackoverflow.com/questions/18479295/
+  // 2. member-initializer-does-not-name-a-non-static-data-member-or-base-class
+  /// Constructor
+  ///
+  /// \param[in] metaSkeletonStateSpace Statespace of the skeleton.
+  /// \param[in] metaskeleton Metaskeleton of the robot.
+  /// \param[in] numIKSolution Number of IK solutions to rank.
+  IKRankingStrategy(
+      statespace::dart::ConstMetaSkeletonStateSpacePtr metaSkeletonStateSpace,
+      ::dart::dynamics::ConstMetaSkeletonPtr metaSkeleton,
+      std::size_t numIKSolutions);
+
+  /// Destructor
+  virtual ~IKRankingStrategy() = default;
+
+  /// Returns the statespace.
+  statespace::ConstStateSpacePtr getStateSpace() const;
```

Comment:
Is this necessary? What do we think we would need the state space for?

Reply:
Oh, is this for the `JointLimitAvoidanceStrategy`?

Reply:
That was the idea, yeah.